### PR TITLE
Eager: miscellaneous fixes

### PR DESCRIPTION
**Pull request by abock**
1. Fix a bug when logging a `DeviceIndex`
2. Update provision-ubuntu-20.04.sh to install torchvision with pip3, used in tests
3. Pin the `data` directory for torchvision data in `mnist_fc_training.py` to be absolute, so it can be run from any `pwd`
4. Don't rebuild ort_aten.g.cpp if codegen didn't change anything
5. Use the execution provider's `CopyTensor` in `CopyVectorToTensor<T>` cc @chandru-r

### CODE REVIEW

**Comment by abock** [comment]
Note: let's merge #47 first since this PR is based on that and carries duplicate commits.

**Comment by abock** [comment]
Rebased since #47 has merged so this one should be easier to review.

**Review by souptc — APPROVED** [review]

File: src/eager/ort_util.h [hunk 1]
```diff
@@ -18,13 +18,34 @@ void CreateMLValue(onnxruntime::AllocatorPtr alloc,
...
+
+  execution_provider.GetDataTransfer()->CopyTensor(
+    ort_value->Get<onnxruntime::Tensor>(),
+    tensor);
 }
 
 // vector<bool> is specialized so we need to handle it separately
 template <>
-inline void CopyVectorToTensor<bool>(const std::vector<bool>& value, onnxruntime::Tensor& tensor) {
+inline void CopyVectorToTensor<bool>(onnxruntime::ORTInvoker& invoker,
```
Comment (chandru-r) [inline src/eager/ort_util.h:46]: We should probably remove this specialization since it will not work when the Tensor is not in CPU.

**Review by chandru-r — APPROVED** [review]

File: src/eager/ort_util.h [hunk 2]
```diff
@@ -18,13 +18,33 @@ void CreateMLValue(onnxruntime::AllocatorPtr alloc,
...
+inline void CopyVectorToTensor(onnxruntime::ORTInvoker& invoker,
+                               const std::vector<T>& value,
+                               onnxruntime::Tensor& tensor) {
+  const auto& execution_provider = invoker.GetCurrentExecutionProvider();
+
+  OrtValue* ort_value;
+  int64_t shape = value.size();
+
+  Ort::ThrowOnError(Ort::GetApi().CreateTensorWithDataAsOrtValue(
+    &execution_provider.GetAllocator(0, OrtMemTypeDefault)->Info(),
```
[the remark this answers is not in the document]
Comment (chandru-r) [reply src/eager/ort_util.h]: I pushed a commit to fix this.

**Pull request merged**